src/adapter/compact.rs

```diff
@@ -0,0 +1,119 @@
+//! Module for use with `#[serde(with = "...")]` to serialize a [`Uuid`]
+//! as a `[u8; 16]
+//!
+//! [`Uuid`]: ../../struct.Uuid.html
+
+use serde::de::{self, Error, SeqAccess};
+use serde::ser::SerializeTuple;
+use serde::{Deserializer, Serializer};
+
+use prelude::*;
+
+use std::fmt;
+
+/// Serializer for a [`Uuid`] into a `[u8; 16]`
+///
+/// [`Uuid`]: ../../struct.Uuid.html
+pub fn serialize<S: Serializer>(
+    u: &Uuid,
+    serializer: S,
+) -> Result<S::Ok, S::Error> {
+    println!("sereeeeeeeeealize");
+    let mut seq = serializer.serialize_tuple(16)?;
+
+    for byte in u.as_bytes() {
+        seq.serialize_element(byte)?;
+    }
+
+    seq.end()
+}
+
+/// Deserializer from a `[u8; 16]` into a [`Uuid`]
+///
+/// [`Uuid`]: ../../struct.Uuid.html
+pub fn deserialize<'de, D: Deserializer<'de>>(
+    deserializer: D,
+) -> Result<Uuid, D::Error> {
+    println!("desereeeeeeeeialize");
+    struct DenseUuidBytesVisitor;
```

Comment:
do we need the struct inside of the function? Generally not a fan of this.

Reply:
I was following in the pattern of the String visitors in the (De)Serialize impls in serde_support.rs, it can be moved if you prefer

Reply:
It is fine. Just something I don't like personally but i'm okay with it if it keeps it consistent with other impls in the crate

Reply:
Huuuuuuggggeee necro...


It just keeps it very much contained and prevents any accidental use outside the function